src/modules/flow/converter/converter.c

```diff
@@ -2066,4 +2067,81 @@ string_to_timestamp_convert(struct sol_flow_node *node, void *data, uint16_t por
     return 0;
 }
 
+static bool
+json_validate(struct sol_blob *blob, enum sol_json_type type)
+{
+    struct sol_json_scanner scanner;
+    struct sol_json_token token;
+
+    sol_json_scanner_init(&scanner, blob->mem, blob->size);
+    return sol_json_scanner_next(&scanner, &token) &&
+           sol_json_token_get_type(&token) == type &&
+           sol_json_scanner_skip_over(&scanner, &token);
```

Comment:
would be too much to check if after skip-over token->end == ((char *)blob->mem + blob->size)? eventually `-1` if there is a trailing null byte in the blob.